src/app/board.service.ts

```diff
@@ -49,6 +53,14 @@ export class BoardService {
     this.currentBoardKey = boardKey;
     const board = this.boardSet.getBoard(this.currentBoardKey);
     this.observer.next(board);
+    this.pageStackService.addPage(boardKey);
+  }
+
+  back() {
+    const boardKey = this.pageStackService.back();
```

Comment:
This probably shouldn't be called back as it isn't an action, maybe get_prev_page or something similar?